common/reportMetrics.py

```diff
@@ -431,20 +431,73 @@ def process_replay(file_path):
-                                if colname.find(":") == -1:
-                                    groupid = i
-                                    colname = colname + ":" + str(groupid)
-                                current_row[colname] = row[i]
+                                column_name = field_names[i].strip()
+                                metric = column_name.split("[")[0]
+                                if column_name.find("]") == -1:
+                                    column_name = column_name + "[-]"
+                                if column_name.find(":") == -1:
+                                    group_id = get_grouping_id(metric, grouping_map)
+                                    column_name = column_name + ":" + str(group_id)
+                                elif len(column_name.split(":")[1]) == 0:
+                                    group_id = get_grouping_id(metric, grouping_map)
+                                    column_name = column_name + str(group_id)
+                                current_row[column_name] = row[i]
                         to_send_metric_data.append(current_row)
                         current_line_count += 1
                 # send final chunk
                 if len(to_send_metric_data) != 0:
                     send_data([to_send_metric_data, min_timestamp_epoch, max_timestamp_epoch], file_path, chunk_count + 1)
                     chunk_count += 1
                 logger.debug("Total chunks created: " + str(chunk_count))
+                save_grouping(grouping_map)
+
+
+def save_grouping(metric_grouping):
+    """
+    Saves the grouping data to grouping.json
+    Parameters:
+        - `grouping_map` : metric_name-grouping_id dict
+    :return: None
+    """
+    with open('grouping.json', 'w+') as f:
+        f.write(json.dumps(metric_grouping))
+
+
+def load_grouping():
+    """
+    Loads the grouping data from grouping.json
+    :return: grouping JSON string
+    """
+    if os.path.isfile('grouping.json'):
+        logger.debug("Grouping file exists. Loading..")
+        with open('grouping.json', 'r+') as f:
+            try:
+                grouping_json = json.loads(f.read())
+            except ValueError:
+                grouping_json = json.loads("{}")
+                logger.debug("Error parsing grouping.json.")
+    else:
+        grouping_json = json.loads("{}")
+    return grouping_json
+
+
+def get_grouping_id(metric_key, metric_grouping):
+    """
+    Get grouping id for a metric key
+    Parameters:
+    - `metric_key` : metric key str to get group id.
+    - `metric_grouping` : metric_key-grouping id map
+    """
+    for index in range(3):
+        grouping_candidate = random.randint(GROUPING_START, GROUPING_END)
+        if metric_key in metric_grouping:
+            grouping_id = int(metric_grouping[metric_key])
+            return grouping_id
+        else:
+            grouping_id = grouping_candidate
```

Comment:
please add dictionary to check if this grouping id already exists. The current logic does not work if the grouping candidate is already existed.